optaplanner-examples/src/main/resources/org/optaplanner/examples/conferencescheduling/solver/conferenceSchedulingScoreRules.drl

```diff
@@ -31,361 +31,405 @@ global HardMediumSoftScoreHolder scoreHolder;
 end
 
 rule "Talk prohibited timeslot tags"
     when
-        Talk($count : prevailingProhibitedTimeslotTagCount() > 0)
+        $talk : Talk($count : prevailingProhibitedTimeslotTagCount() > 0)
     then
-        scoreHolder.penalize(kcontext, $count);
+        scoreHolder.penalize(kcontext,
+                $count * $talk.getDurationInMinutes());
 end
 
 rule "Speaker required room tags"
     when
-        Talk($count : missingSpeakerRequiredRoomTagCount() > 0)
+        $talk : Talk($count : missingSpeakerRequiredRoomTagCount() > 0)
     then
-        scoreHolder.penalize(kcontext, $count);
+        scoreHolder.penalize(kcontext,
+                $count * $talk.getDurationInMinutes());
 end
 
 rule "Speaker prohibited room tags"
     when
-        Talk($count : prevailingSpeakerProhibitedRoomTagCount() > 0)
+        $talk : Talk($count : prevailingSpeakerProhibitedRoomTagCount() > 0)
     then
-        scoreHolder.penalize(kcontext, $count);
+        scoreHolder.penalize(kcontext,
+                $count * $talk.getDurationInMinutes());
 end
 
 rule "Talk required room tags"
     when
-        Talk($count : missingRequiredRoomTagCount() > 0)
+        $talk : Talk($count : missingRequiredRoomTagCount() > 0)
     then
-        scoreHolder.penalize(kcontext, $count);
+        scoreHolder.penalize(kcontext,
+                $count * $talk.getDurationInMinutes());
 end
 
 rule "Talk prohibited room tags"
     when
-        Talk($count : prevailingProhibitedRoomTagCount() > 0)
+        $talk : Talk($count : prevailingProhibitedRoomTagCount() > 0)
     then
-        scoreHolder.penalize(kcontext, $count);
+        scoreHolder.penalize(kcontext,
+                $count * $talk.getDurationInMinutes());
 end
 
-rule "Talk prerequisite talks"
-    when
-        $talk : Talk(timeslot != null, getPrerequisiteTalkSet().size() > 0)
-        $count : Number(this > 0) from accumulate(
-            $prerequisiteTalk : Talk(timeslot != null,
-                    $talk.getPrerequisiteTalkSet().contains(this),
-                    !getTimeslot().endsBefore($talk.getTimeslot())),
-            count($prerequisiteTalk)
-        )
-    then
-        scoreHolder.penalize(kcontext, $count.intValue());
-end
+// ############################################################################
+// Medium constraints
+// ############################################################################
 
-rule "Consecutive talks pause"
+rule "Published timeslot"
     when
-        ConferenceConstraintConfiguration($minimumPause : minimumConsecutiveTalksPauseInMinutes)
-        $talk: Talk(timeslot != null, $id : id, $timeslot : getTimeslot())
-        Talk(timeslot != null, $id < id, hasMutualSpeaker($talk), !getTimeslot().pauseExists($timeslot, $minimumPause))
+        $talk : Talk(publishedTimeslot != null, timeslot != publishedTimeslot)
     then
-        scoreHolder.penalize(kcontext);
+        scoreHolder.penalize(kcontext, $talk.getDurationInMinutes());
```

Comment:
This will penalize 1 change of 2hr talk more than 3 changes of 30m talks (which possibly disrupts 3 speakers' schedules instead of only one). I think for this constraint it's better to penalize equally (or based on the number of speakers), wdyt?